kratos/mpi/utilities/amgcl_distributed_csr_conversion_utilities.h

```diff
@@ -0,0 +1,167 @@
+
+//    |  /           | 
+//    ' /   __| _` | __|  _ \   __| 
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/ 
+//                   Multi-Physics  
+//
+//  License:		 BSD License 
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Denis Demidov
+//                   Riccardo Rossi
+//                   
+//
+
+#if !defined(KRATOS_DISTRIBUTED_CSR_CONVERSION_UTILITIES_H_INCLUDED)
+#define  KRATOS_DISTRIBUTED_CSR_CONVERSION_UTILITIES_H_INCLUDED
+
+// System includes
+
+// External includes
+#include <amgcl/backend/builtin.hpp>
+#include <amgcl/backend/interface.hpp>
+#include <amgcl/mpi/distributed_matrix.hpp>
+#include <amgcl/adapter/zero_copy.hpp>
+
+// Project includes
+#include "containers/distributed_csr_matrix.h"
+#include "utilities/amgcl_csr_conversion_utilities.h"
+#include "mpi/includes/mpi_data_communicator.h"
+
+namespace Kratos
+{
+
+/**
+Utilities to convert the distributed_csr matrix to other libraries
+ */
+class AmgclDistributedCSRConversionUtilities
+{
+
+public:
+
+    /**
+     This function returns a shared pointer to an Amgcl distributed_matrix
+     */
+	template< class TDataType, class TIndexType >
+	static Kratos::shared_ptr<amgcl::mpi::distributed_matrix<amgcl::backend::builtin<double>>> ConvertToAmgcl(
+			const DistributedCsrMatrix<TDataType, TIndexType>& rA,
+			DenseVector<TIndexType>& global_index2,
+			bool MoveToBackend=true)
+	{
+		IndexType chunk = rA.local_size1();
+
+    	auto loc_a = amgcl::adapter::zero_copy(chunk, 
+                rA.GetDiagonalBlock().index1_data().begin(),
+                rA.GetDiagonalBlock().index2_data().begin(),
+                rA.GetDiagonalBlock().value_data().begin()
+			);
+		loc_a->ncols = rA.GetDiagonalBlock().size2(); //important if the matrix is not square
+
+		auto rem_a = amgcl::adapter::zero_copy(chunk, 
+			rA.GetOffDiagonalBlock().index1_data().begin(),
+			global_index2.data().begin(),
+			rA.GetOffDiagonalBlock().value_data().begin()
+		);
+		rem_a->ncols = rA.GetOffDiagonalBlock().size2(); //important if the matrix is not square
+
+		auto raw_mpi_comm = MPIDataCommunicator::GetMPICommunicator( rA.GetComm());
+		amgcl::mpi::communicator comm(raw_mpi_comm);
+
+		auto pAmgcl = Kratos::make_shared<amgcl::mpi::distributed_matrix<amgcl::backend::builtin<double>>>(comm, loc_a, rem_a);
+
+		if(MoveToBackend)
+			pAmgcl->move_to_backend();
+
+		return pAmgcl;
+	}
+
+	template< class TDataType, class TIndexType >
+	static typename DistributedCsrMatrix<TDataType, TIndexType>::UniquePointer ConvertToCsrMatrix(
+			amgcl::mpi::distributed_matrix<amgcl::backend::builtin<double>>& rA, //cannot be made const since i need to modify some data in-place,
+			DataCommunicator& kratos_comm=ParallelEnvironment::GetDefaultDataCommunicator()
```

Comment:
```suggestion
			DataCommunicator& kratos_comm
```

this is dangerous, can you please remove it?

Reply:
well, i would say that it is a fair policy to pass the default, as long as a user can change it as he wishes. Having said this it is little to no work to change it so i have done it